Source/DafnyCore/Compilers/CoverageInstrumenter.cs

```diff
@@ -82,4 +93,20 @@ public class CoverageInstrumenter {
       legend = null;
     }
   }
+
+  public void PopulateCoverageReport(CoverageReport coverageReport) {
+    var coverageReportDir = compiler.Options?.Get(CommonOptionBag.ExecutionCoverageReport);
+    if (coverageReportDir != null) {
+      var tallies = File.ReadLines(talliesFilePath).Select(int.Parse).ToArray();
+      foreach (var ((token, _), tally) in legend.Zip(tallies)) {
+        var label = tally == 0 ? CoverageLabel.NotCovered : CoverageLabel.FullyCovered;
+        // For now we only identify branches at the line granularity,
+        // which matches what `dafny generate-tests ... --coverage-report` does as well.
```

Comment:
This is nice because then merging actual and expected coverage would work well. The `merge-coverage-reports` command could perhaps use a `--diff` flag that would produce a report indicating where two coverage reports differed.

Reply:
My thoughts exactly, and although I do suspect that the actual and expected coverage don't 100% agree on coverable things yet, I don't think it will take much to align them in the future.